src/Compilers/CSharp/Portable/Binder/Semantics/OverloadResolution/OverloadResolution.cs

```diff
@@ -1032,34 +1083,51 @@ private void RemoveWorseMembers<TMember>(ArrayBuilder<MemberResolutionResult<TMe
             // then we need to do no more work; we know it cannot win. But it is also possible that
             // it is not worse than anything but not better than everything. 
 
+            if (results.Count <= 1)
```

Comment:
> Can we do better here? Something like GetValidCount for methods?

There is already RemainingCandidatesCount helper and one consumer of the RemoveWorseMembers method uses it to decide if RemoveWorseMembers should be called. However it only checks for null. We should probably make sure that all callers never call RemoveWorseMembers if RemainingCandidatesCount returns a value less than 2. Then this check will be unnecessary.